editor/gui/progress_dialog.cpp

```diff
@@ -168,14 +168,14 @@ void ProgressDialog::_popup() {
 
 	Window *current_window = SceneTree::get_singleton()->get_root()->get_last_exclusive_window();
 	ERR_FAIL_NULL(current_window);
-	reparent(current_window);
+	callable_mp((Control *)this, &Control::reparent).call_deferred(current_window, true);
 
 	// Ensures that events are properly released before the dialog blocks input.
 	bool window_is_input_disabled = current_window->is_input_disabled();
 	current_window->set_disable_input(!window_is_input_disabled);
 	current_window->set_disable_input(window_is_input_disabled);
 
-	show();
+	callable_mp((CanvasItem *)this, &CanvasItem::show).call_deferred();
```

Comment:
Why this needs to be deferred?

Reply:
To avoid showing the dialog before `reparent`, is it ok to make it visible first then changing its parent ? I just have tried to avoid any further issues that may happen because of this.

Reply:
Showing it earlier shouldn't cause problems, but deferring makes sense.
It would be nice if the 2 deferred calls were done together, either just moved to be one after another, or as part of a separate method (like `_reparent_and_show()` which would then be deferred instead).